presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -173,4 +175,36 @@ private MethodHandle applyExtraParameters(Method matchingMethod, List<Object> ex
         }
         return clazz;
     }
+
+    static final class PolymorphicScalarFunctionChoice
```

Comment:
private

Reply:
I can not make this  private. It needs to stay package-private since PolymorphicScalarFunctionBuilder is using it.

Reply:
I will move MethodsGroup to PolymorphicScalarFunction because it is a member of PolymorphicScalarFunctionChoice. PolymorphicScalarFunctionBuilder should contain builders only.